centos/httpdir/centos-7.cfg.tpl

```diff
@@ -24,4 +24,9 @@ openssh-server
 %end
 %post
 /usr/bin/yum -y install sudo
+sed -i "s/PermitRootLogin .*/PermitRootLogin yes/" /etc/ssh/sshd_config
+sed -i "s/^#[\s]*PermitRootLogin .*/PermitRootLogin yes/" /etc/ssh/sshd_config
```

Comment:
Not sure what you exactly want to match, but can't you use a single sed call like this one that should match commented and non commented strings with different number of whitespaces used:

`sed 's/^[#[:space:]]*PermitRootLogin .*/PermitRootLogin yes/' /etc/ssh/sshd_config` ?

Using `[:space:]` should be also more posix-compliant (see [sed](https://pubs.opengroup.org/onlinepubs/009695399/utilities/sed.html) and [regex](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap09.html#tag_09_03) and should work with both macos and CentOS's sed versions.

Reply:
Fixed as suggested